src/tests/class_tests/openms/source/MSSpectrum_test.cpp

```diff
@@ -72,6 +72,10 @@ MSSpectrum* nullPointer = nullptr;
 START_SECTION((MSSpectrum()))
   ptr = new MSSpectrum();
   TEST_NOT_EQUAL(ptr, nullPointer)
+
+  // Ensure that MSSpectrum has a no-except move constructor (otherwise
+  // std::vector is inefficient and will copy instead of move).
+  TEST_EQUAL(noexcept(MSSpectrum(std::declval<MSSpectrum&&>())), true)
```

Comment:
could this be implemented using a static_assert in the class itself?

Reply:
> could this be implemented using a static_assert in the class itself?

yes, except MSVS 2015 will not work on this: https://godbolt.org/z/utMmsl (MSVS 2017 will work)

Reply:
static_assert(noexcept(M(std::declval<M&&>())), "failed" ); works for me

Reply:
> could this be implemented using a static_assert in the class itself?

I just discussed this with Timo and we probably would want to do this as a macro. There are two places where this could go, either in the class or in the test file. There are pros and cons to both, of course it increases bloat if it goes into the class itself but it will also ensure that every future compilation of OpenMS will ensure that this is true and it adds to the "documentation" if every client can see how the class behaves. On the other hand, we do all of our testing in specialized files and this seems a logical place to do testing about assumptions of the class. We may actually end up testing more than just this, see for example https://philippegroarke.com/posts/2018/easy_defensive_programming/